components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java

```diff
@@ -1091,19 +1093,38 @@ public Response apisChangeLifecyclePost(String action, String apiId, String life
                 //since workflows are not defined for checklist items
                 workflowResponse.setWorkflowStatus(WorkflowStatus.APPROVED);
                 response = MappingUtil.toWorkflowResponseDTO(workflowResponse);
+                return Response.ok().entity(response).build();
             } else {
                 WorkflowResponse workflowResponse = RestAPIPublisherUtil.getApiPublisher(username)
                         .updateAPIStatus(apiId, action, lifecycleChecklistMap);
                 response = MappingUtil.toWorkflowResponseDTO(workflowResponse);
+                //if workflow is in pending state or if the executor sends any httpworklfowresponse (workflow state can 
+                //be in either pending or approved state) send back the workflow response 
+                if (WorkflowStatus.CREATED == workflowResponse.getWorkflowStatus()
+                        || workflowResponse instanceof HttpWorkflowResponse) {
+                    URI location = new URI(RestApiConstants.RESOURCE_PATH_APIS + "/" + apiId);
+                    return Response.status(Response.Status.ACCEPTED).header(RestApiConstants.LOCATION_HEADER, location)
```

Comment:
I think its better to use Response.created(location) to set the location header. Please check if this is re available at msf4j 2.3.0-m2, and update it